UI/window-basic-main.cpp

```diff
@@ -231,6 +231,38 @@ static void AddExtraModulePaths()
 #endif
 }
 
+static const char *safe_modules[] = {
+	// Cross-platform
+	"aja", "aja-output-ui", "decklink", "decklink-captions",
+	"decklink-output-ui", "frontend-tools", "image-source", "obs-browser",
+	"obs-browser-page", "obs-ffmpeg", "obs-filters", "obs-libfdk",
+	"obs-outputs", "obs-transitions", "obs-vst", "obs-websocket",
+	"obs-x264", "rtmp-services", "text-freetype2", "vlc-video",
+#ifdef _WIN32
+	// Windows
+	"coreaudio-encoder", "obs-qsv11", "obs-text", "win-capture",
+	"win-dshow", "win-wasapi",
+#elif defined(__APPLE__)
+	// macOS
+	"coreaudio-encoder", "mac-avcapture", "mac-capture", "mac-syphon",
+	"mac-videotoolbox", "mac-virtualcam",
+#else
+	// Linux (and BSDs...)
+	"linux-alsa", "linux-capture", "linux-jack", "linux-pipewire",
+	"linux-pulseaudio", "linux-v4l2", "oss-audio", "sndio",
+#endif
+	NULL};
```

Comment:
This should be generated through CMake in a generated header.
Maybe a macro with all the string and add it to `safe_modules`, and the macro is CMake generated.

Reply:
Updated to create the list from the CMake `OBS_MODULES_LIST` variable.